go/controller/internal/controller/managednamespace_controller.go

```diff
@@ -308,10 +304,6 @@ func (r *ManagedNamespaceReconciler) getProject(ctx context.Context, ns *v1alpha
 	project := &v1alpha1.Project{}
 	if ns.Spec.ProjectRef != nil {
 		if err := r.Get(ctx, client.ObjectKey{Name: ns.Spec.ProjectRef.Name}, project); err != nil {
-			if errors.IsNotFound(err) {
```

Comment:
this should wait for resources

Reply:
it does in `handleRequeue` method. We simplified the whole logic to one method. We don't need extra checks

Reply:
ahhh ok